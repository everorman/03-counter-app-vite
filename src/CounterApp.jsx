import PropTypes from 'prop-types';
import { useState } from 'react';

// eslint-disable-next-line react/prop-types
export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);
  const handleAdd = () => {
    setCounter((c) => c + 1);
  };
  const handleSubtract = () => {
    setCounter((c) => c - 1);
  };
  const handleReset = () => {
    setCounter(value);
  };
  return (
    <>
      <h1>Counter App</h1>
      <h2> {counter} </h2>
      <button onClick={handleAdd}>+1</button>
      <button onClick={handleSubtract}>-1</button>
      <button onClick={handleReset}>Reset</button>
    </>
  );
};

CounterApp.propsType = {
  value: PropTypes.number,
};

CounterApp.defaultProps = {
  value: 1,
};
