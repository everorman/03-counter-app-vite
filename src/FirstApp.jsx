import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle }) => {
  return (
    <>
      <h1 data-testid='test-title'>{title}</h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  subTitle: 'Evert Ortiz',
  title: 'Aprendiendo react',
};
