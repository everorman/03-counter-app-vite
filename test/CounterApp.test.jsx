import { CounterApp } from '../src/CounterApp';
import { render, screen, fireEvent, getByText } from '@testing-library/react';

describe('CounterApp', () => {
  const initialValue = 100;
  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<CounterApp />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar el valor inicial 100', () => {
    render(<CounterApp value={initialValue} />);
    expect(screen.getByRole('heading', { level: 2 }).innerHTML).toContain(initialValue.toString());
  });
  test('Debe incrementar 1 al hacer click', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('+1'));
    expect(screen.getByText('101')).toBeTruthy();
  });
  test('Debe decrementar 1 al hacer click', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('-1'));
    expect(screen.getByText('99')).toBeTruthy();
  });
  test('Debe implementar el restart', () => {
    render(<CounterApp value={initialValue} />);
    fireEvent.click(screen.getByText('-1'));
    fireEvent.click(screen.getByText('-1'));
    fireEvent.click(screen.getByText('Reset'));
    expect(screen.getByText('100')).toBeTruthy();
  });
});
