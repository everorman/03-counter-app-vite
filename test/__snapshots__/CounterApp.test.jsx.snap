// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterApp Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    Counter App
  </h1>
  <h2>
     
    1
     
  </h2>
  <button>
    +1
  </button>
  <button>
    -1
  </button>
  <button>
    Reset
  </button>
</div>
`;
